### PR TITLE
nomad: add bash completion

### PR DESCRIPTION
**Pull request by cdepillabout**
###### Description of changes

This PR adds bash completion to Nomad.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/nomad/default.nix
Comment (cdepillabout) [inline pkgs/applications/networking/cluster/nomad/default.nix:37]: I just copied this from how Vault installs bash completion:

https://github.com/NixOS/nixpkgs/blob/1eaf23e8febf68394df1e6d3304c7a02dbf4c59c/pkgs/tools/security/vault/default.nix#L33-L35

**Review by wraithm — APPROVED** [review]

**Review by teutat3s — APPROVED** [review]

**Comment by cdepillabout** [comment]
This PR has gotten two approvals, and it is a relatively straightforward change, so I'm going to go ahead and merge it in.

If any of the maintainers end up seeing problems with this, please feel free to leave a comment and I'd be happy to send a follow-up PR fixing things up.

**Pull request merged**
